### PR TITLE
Convert rst files to md

### LINKED ISSUE #155021: Convert to markdown: fsdp.rst, func.api.rst, func.batch_norm.rst, func.migrating.rst, func.rst
Convert the following files to MyST markdown and ensure that the doc tests are green:

- [fsdp.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/fsdp.rst)
- [func.api.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/func.api.rst)
- [func.batch_norm.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/func.batch_norm.rst)
- [func.migrating.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/func.migrating.rst)
- [func.rst](https://github.com/pytorch/pytorch/tree/main/docs/source/func.rst)

To convert an `.rst` file:

1) Assign the issue to yourself using the `/assigntome` comment.

2) **Important!** Use `git mv filename.rst filename.md` to convert the issue from `.rst` to `.md`. **DO NOT delete**
   the `.rst` and create the `.md` - we won't accept PRs with deleted `.rst` files. We want to preserve the git log of
   those files. Therefore, using `git mv` is very important!

3) Before submitting a PR, build the docs locally as described in the [README](https://github.com/pytorch/pytorch/blob/main/README.md#building-the-documentation).
   Preview localy with `make serve` to ensure that the HTML renders correctly.    

4) Run `lintrunner` from the root of the repository:

   ```
   pip install lintrunner
   lintrunner init
   lintrunner -m main
   ```

4) Submit a PR with your changes and ensure the PR passes the doc CI tests. Specifically it needs to pass:

   * pull / linux-docs / build-docs-python-false (pull_request)
   * linux-jammy-py3.9-gcc11 / test (docs_test, 1, 1, lf.linux.2xlarge)

   Doc tests start after the Docker build, which might take about 30 minutes. 
   The doc build itself should take about 25 minutes. When the build is done, 
   you can see the preview of your build by clicking:
   📄 **Preview Python docs built from this PR** in the `pytorch-bot` comment.

   Make sure your PR is tagged with `docathon-h1-2025`, `module: docs`, `topic: not user facing`,
   and `topic: docs` labels.

5) Compare the preview of the pages you converted with same pages it the main branch.
   The converted pages in preview must look the same as in the main branch.

6) Once the PR is accepted, one of the maintainers will merge it.

See [Quick MyST Conversion Reference](https://github.com/pytorch/pytorch/wiki/Docstring-Guidelines#restructuredtext-vs-markdown)
in our wiki or [MyST Parser documentation](https://myst-parser.readthedocs.io/en/v0.15.1/sphinx/use.html) for detailed syntax.

Here is an example [PR](https://github.com/pytorch/pytorch/pull/154438)

cc @sekyondaMeta @AlannaBurke

### LINKED ISSUE #155158: ERROR: Unknown target name: "deepspeed"
### 📚 The doc issue

while building the docs I found the following issue:

```
/localdisk/jafraust/miniforge/envs/docs/lib/python3.10/site-packages/torch/distributed/fsdp/fully_sharded_data_parallel.py:docstring of torch.distributed.fsdp.fully_sharded_data_parallel.FullyShardedDataParallel:3: ERROR: Unknown target name: "xu et al.".
/localdisk/jafraust/miniforge/envs/docs/lib/python3.10/site-packages/torch/distributed/fsdp/fully_sharded_data_parallel.py:docstring of torch.distributed.fsdp.fully_sharded_data_parallel.FullyShardedDataParallel:3: ERROR: Unknown target name: "deepspeed".
looking for now-outdated files... none found
```

### Suggest a potential alternative/fix

change the format from this

```
    This is inspired by `Xu et al.`_ as well as the ZeRO Stage 3 from DeepSpeed_.
    FullyShardedDataParallel is commonly shortened to FSDP.

    .. _`Xu et al.`: https://arxiv.org/abs/2004.13336
    .. _DeepSpeed: https://www.deepspeed.ai/
````

to

```
his is inspired by `Xu et al. <https://arxiv.org/abs/2004.13336>`_ as well as the ZeRO Stage 3 from `DeepSpeed <https://www.deepspeed.ai/>`_.
    FullyShardedDataParallel is commonly shortened to FSDP.
```

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @svekars @sekyondaMeta @AlannaBurke

### PR DESCRIPTION
**Pull request by jafraustro**
Fixes #155021 
Fixes #155158 

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @svekars @sekyondaMeta @AlannaBurke

### CODE REVIEW

**Comment by jafraustro** [comment]
Preview links to the pages:

fsdp.rst: https://docs-preview.pytorch.org/pytorch/pytorch/155369/fsdp.html#module-torch.distributed.fsdp
func.api.rst: https://docs-preview.pytorch.org/pytorch/pytorch/155369/func.api.html
func.batch_norm.rst: https://docs-preview.pytorch.org/pytorch/pytorch/155369/func.batch_norm.html
func.migrating.rst: https://docs-preview.pytorch.org/pytorch/pytorch/155369/func.migrating.html
func.rst: https://docs-preview.pytorch.org/pytorch/pytorch/155369/func.html

File: docs/source/func.api.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# torch.func API Reference
```
Comment (svekars) [inline docs/source/func.api.md:1]: Created on date for this file has changed:[ func.api.html ](https://docs-preview.pytorch.org/pytorch/pytorch/155369/func.api.html#module-torch.func)(compared to [main](https://docs.pytorch.org/docs/main/func.api.html) - this needs to be fixed.

File: docs/source/func.migrating.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
+# Migrating from functorch to torch.func
```
Comment (svekars) [inline docs/source/func.migrating.md:1]: Created On date for this file has changed https://docs-preview.pytorch.org/pytorch/pytorch/155369/func.migrating.html compared to [main](https://docs.pytorch.org/docs/main/func.migrating.html) - this has to be fixed.

Comment (jafraustro) [reply docs/source/func.migrating.md:1]: I split the one commit into three commits, so the use of `git mv` is clearly specified.

File: docs/source/func.api.md (hunk 1 above)
Comment (jafraustro) [reply docs/source/func.api.md:1]: I split the one commit into three commits, so the use of git mv is clearly specified.

**Review by svekars — APPROVED** [review]

**Review by malfet — APPROVED** [review]
